### PR TITLE
Provide instructions on the PR workflow based on OWNERS files.

### LINKED ISSUE #3: Setup kubeflow.org domain
We need to setup the kubeflow.org domain.

The domain was purchased by @DjangoPeng and Caicloud and contributed to the community. Thank you very much.

The domain has been transferred to Google Domains to facilitate setup.

Current owners:
@DjangoPeng (CaiCloud)
@jlewi (Google)
@aronchick (Google)

There were a bunch of custom resource records defined
custom_resource_records

These look like vestiges of using the original domain registrar. @DjangoPeng  can we delete these?

### PR DESCRIPTION
**Pull request by jlewi**
* Instructions describe how we will use prow bots and OWNERs files
to manage the PR workload.

* Instructions are based on the Kubernetes workflow
https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#code-review-using-owners-files

Fix #3

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/community/27)

### CODE REVIEW

**Comment by jlewi** [comment]
@DjangoPeng @gaocegege @erikerlandson @ddysher Does someone want to review this?

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  - adds the  `approved` label once an **approver** for each of the required
+    OWNERS files has `/approve`'d
+  - comments as required OWNERS files are satisfied
+  - removes outdated approval status comments
+- [plugin: blunderbuss](https://git.k8s.io/test-infra/prow/plugins/assign)
+  - determines **reviewers** and requests their reviews on PR's
+- [plugin: lgtm](https://git.k8s.io/test-infra/prow/plugins/lgtm)
+  - adds the `lgtm` label when a **reviewer** comments `/lgtm` on a PR
+  - the **PR author** may not `/lgtm` their own PR
+- [pkg: k8s.io/test-infra/prow/repowners](https://git.k8s.io/test-infra/prow/repoowners/repoowners.go)
```
Comment (ScorpioCPH) [inline contributing.md]: typo: s/repowners/repoowners

**Comment by jlewi** [comment]
@ScorpioCPH Changes from the K8s doc are trivially. Main difference is "kubernetes" -> "kubeflow" and I deleted the section about mungeithub.

Don't forget to "/lgtm"

**Comment by gaocegege** [comment]
Thanks @k8s-ci-robot :smile:

**Pull request merged**